### PR TITLE
feat: update Comment By ID Endpoint

### PR DESCRIPTION
**Pull request by chukssomzzy**
## Description
This feature allows updating a comment by its ID. It facilitates the modification of any comment attached to any model by specifying the comment ID.

## Related Issue
[FEAT: Update Comment by ID](https://github.com/hngprojects/hng_boilerplate_python_fastapi_web/issues/362)

## Motivation and Context
This change is necessary to provide functionality for updating comments, enhancing the flexibility and usability of the application. It solves the problem of not being able to edit existing comments.

## How Has This Been Tested?
- [X] Tested extensively with Postman.
- [X] Written working unittests with pytest.

## Screenshots (if appropriate):
image

## Types of changes
- [X] Bug fix (non-breaking change which fixes an issue).
- [X] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to change).

## Checklist:
- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.
- [X] I have read the **CONTRIBUTING** document.
- [X] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by chukssomzzy** [comment]
@Goketech

**Comment by chukssomzzy** [comment]
@Goketech

**Review by Goketech — DISMISSED** [review]

**Comment by chukssomzzy** [comment]
@joboy-dev

**Review by Goketech — APPROVED** [review]

**Review by Goketech — APPROVED** [review]

**Review by Goketech — APPROVED** [review]

**Review by joboy-dev — DISMISSED** [review]

**Review by joboy-dev — DISMISSED** [review]

**Review by Goketech — APPROVED** [review]

File: test.py [hunk 1]
```diff
@@ -0,0 +1 @@
+jfadfahfadflah
```
Comment (Goketech) [inline test.py:1]: i guess you added this by mistake

**Review by Goketech — CHANGES_REQUESTED** [review]

File: test.py (hunk 1 above)
Comment (chukssomzzy) [reply test.py]: that was a mistake, I have removed and fixed conflict

**Comment by chukssomzzy** [comment]
@zxenonx please review.

**Review by Goketech — APPROVED** [review]

**Review by joboy-dev — APPROVED** [review]

**Review by joboy-dev — APPROVED** [review]

**Review by joboy-dev — CHANGES_REQUESTED** [review]
Resolve conflicts

**Review by CodewithSegNet — CHANGES_REQUESTED** [review]
Work on the response body.. follow the format given to you and add the new screenshot to the PR

**Pull request closed without merging**